src/main/webapp/app/guided-tour/guided-tour-step.model.ts

```diff
@@ -31,12 +13,58 @@ export interface TourStep {
     skipStep?: boolean;
     /** Adds some padding for things like sticky headers when scrolling to an element */
     scrollAdjustment?: number;
-    /** Adds default padding around tour highlighting. Does not need to be true for highlightPadding to work */
-    useHighlightPadding?: boolean;
     /** Adds padding around tour highlighting in pixels, this overwrites the default for this step. Is not dependent on useHighlightPadding being true */
     highlightPadding?: number;
     /** Permission to view step, if no permission is set then the tour step is visible to ROLE_USER
      * Possible inputs: 'ROLE_ADMIN', 'ROLE_INSTRUCTOR', 'ROLE_TA'
      */
     permission?: string[];
 }
+
+export class TextTourStep extends TourStep {
+    /** Translation key for the title **/
+    headlineTranslateKey: string;
+    /** Translation key for the title **/
+    subHeadlineTranslateKey?: string;
+    /** Translation key for the content **/
+    contentTranslateKey: string;
+
+    constructor(tourStep: TextTourStep) {
+        super();
+        Object.assign(this, tourStep);
+    }
+}
+
+export class TextLinkTourStep extends TextTourStep {
+    /** External url **/
+    externalUrl: string;
+    /** Text for external url **/
+    externalUrlTranslateKey: string;
+    /** Link type, either link or button **/
+    linkType: LinkType;
+
+    constructor(tourStep: TextLinkTourStep) {
```

Comment:
As far as I see it the settings of the TourSteps are pure data objects that don't carry functionality.
The repeated constructors could be avoided with a different pattern using generics and settings objects.

```
interface TourStepSettings {
    someSetting: number;
}

interface TextTourStepSettings extends TourStepSettings {
    textStepExclusive: string;
}

class TourStep<T extends TourStepSettings> {
    constructor(public settings: T) {}
}

// These two are equivalent:
const normalStep = new TourStep({someSetting: 3});
const normalStepWithType = new TourStep<TourStepSettings>({someSetting: 3});

// Extending step
const textStep = new TourStep<TextTourStepSettings>({ someSetting: 2, textStepExclusive: 'test'});

// Not allowed, wrong property abc
const incorrectStep = new TourStep<TextTourStepSettings>({ someSetting: 2, abc: 'test'});
```

It would also be great to tie this to an enum somehow so you can easily type check your object in the Angular templates.

Reply:
@thilo-behnke Thank you for you input!
@krusche What do you think about this approach? Should I implement it or do you prefer the current implementation?

Reply:
I like the idea of @thilo-behnke, but I would keep it as it is for now, we can still think about such improvements in one of the next PRs.